lib/resolver.js

```diff
@@ -2,41 +2,48 @@ const EventEmitter = require('events');
 const { getZelFile } = require('./repository');
 
 module.exports = class Resolver extends EventEmitter {
-    /**
-     * @param {Object<failFast>} failFast - Terminates if an error/invalid event occurs
-     */
     constructor(options) {
         super();
+        this.valid = [];
+        this.invalid = [];
         this.opts = options || {};
+        this.fetch = this.fetch.bind(this);
     }
 
     /**
-     * Validates the repositories
+     * Validates all repositores.
+     *
+     * If all specified repositories are valid, returns a resolved
+     * Promise that resolves a list of config objects for each
+     * repository.
+     *
+     * If any repository is invalid, returns a rejected Promise
+     * of the list of invalid repositories.
      *
      * @param {Array<String>} - List of repositories
+     * @return {Promise<Array<Object>>} - Resolves a list of valid/invalid config objects
      */
     validate(repos) {
-        const failFast = this.opts.failFast;
-
         if (!repos || !repos.length) {
-            this.emit('error', 'No repositories specified.');
-            if (failFast) {
-                return false;
-            }
+            return Promise.reject('No repositories specified.');
         }
 
-        if (failFast) {
-            const isValid = (name) => this.emit('valid', name);
-            Promise.each(repos, getZelFile)
-                .then((repoNames) => repoNames.forEach(isValid))
-                .catch((repoName) => this.emit('invalid', repoName));
-            return;
-        }
-
-        const check = (repoName) => getZelFile(repoName)
```

Comment:
Since we're now resolving before cloning, I'm planning to just pass this list of resolved config objects directly to the cloning script so we can avoid another fetch later.

I was debating whether I should kept that inline `check` function to avoid adding another method. Just like my methods small 😄